account_bank_statement_import_oca_camt54/models/account_bank_statement_import.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2019 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import logging
+
+from odoo import models
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountBankStatementImport(models.TransientModel):
+    """Add process_camt method to account.bank.statement.import."""
+
+    _inherit = "account.bank.statement.import"
+
+    def _create_bank_statements(self, stmts_vals):
+        """ Set balance_end_real if not already provided by the file."""
+
+        statement_line_ids, notifications = super()._create_bank_statements(stmts_vals)
+        statements = self.env["account.bank.statement"].search(
+            [("line_ids", "in", statement_line_ids)]
+        )
+        for statement in statements:
+            if not statement.balance_end_real:
+                amount = sum(statement.line_ids.mapped("amount"))
+                statement.balance_end_real = statement.balance_start + amount
+        return statement_line_ids, notifications
+
+    def _complete_stmts_vals(self, stmts_vals, journal, account_number):
+        """Search partner from partner reference"""
+        stmts_vals = super()._complete_stmts_vals(stmts_vals, journal, account_number)
+        for st_vals in stmts_vals:
```

Comment:
you can't do a mapped() ?

Reply:
I don't really how it can be optimized 
stmts_vals will be returned as a result eventually and this is just dict not orm model